upstairs/src/client.rs

```diff
@@ -1429,10 +1267,13 @@ impl DownstairsClient {
                     });
                 }
                 self.repair_addr = Some(repair_addr);
-                // Nothing to do here, return a marker that tells the Upstairs
-                // to promote if it's time.
-                *state = NegotiationState::WaitActive;
-                Ok(NegotiationResult::WaitActive)
+                *state = NegotiationState::WaitForPromote;
+                self.send(Message::PromoteToActive {
+                    upstairs_id: self.cfg.upstairs_id,
+                    session_id: self.cfg.session_id,
+                    gen: self.cfg.generation(),
+                });
+                Ok(NegotiationResult::NotDone)
```

Comment:
non blocking: 

I was about to leave a comment in the previous PR about how this return value changed from `NotDone` to `WaitActive`, but see that it's reverted here :)

Although I'm now wondering if `NotDone` can be removed, it doesn't seem like a useful variant. Though that'd likely require more work than I think.